webapp/src/main/java/com/box/l10n/mojito/service/image/S3FallbackImageService.java

```diff
@@ -0,0 +1,54 @@
+package com.box.l10n.mojito.service.image;
+
+import com.box.l10n.mojito.entity.Image;
+import org.slf4j.Logger;
+
+import java.util.Optional;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Service that retrieves images from an S3 bucket and falls back to the Mojito DB for retrieval if
+ * an image is not available in the S3 bucket.
+ *
+ * If a requested image cannot be found in S3 an attempt will be made to retrieve from the Mojito DB, if the image is
+ * found in the DB then an async task will be executed to upload the image to S3.
+ *
+ * @author maallen
+ */
+public class S3FallbackImageService implements ImageService {
+
+    static Logger logger = getLogger(S3FallbackImageService.class);
+
+    S3ImageService s3ImageService;
+
+    DatabaseImageService databaseImageService;
+
+    S3UploadImageAsyncTask s3UploadImageAsyncTask;
+
+    public S3FallbackImageService(S3ImageService s3ImageService, DatabaseImageService databaseImageService,
+                                  S3UploadImageAsyncTask s3UploadImageAsyncTask) {
+        this.s3ImageService = s3ImageService;
+        this.databaseImageService = databaseImageService;
+        this.s3UploadImageAsyncTask = s3UploadImageAsyncTask;
+    }
+
+    @Override
+    public Image getImage(String name) {
+        logger.debug("Attempt image retrieval from S3 with name: {}", name);
+        Image image = Optional.ofNullable(s3ImageService.getImage(name)).orElseGet(() ->
```

Comment:
make the ImageService return an Optional? you did the change on the repository interface but not the service one, I guess it would simplify here too.

Eventually, we could even use project reactor to get retries and fallback logic (switchIfEmpty) out of the box.